tests/provider: Kinesis Video not supported in GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```
TestAccAWSKinesisVideoStream_basic
TestAccAWSKinesisVideoStream_disappears
TestAccAWSKinesisVideoStream_options
TestAccAWSKinesisVideoStream_Tags
```

With this error: `no such host`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_kinesis_video_stream

### Terraform Configuration Files

Example config:

```hcl
resource "aws_kinesis_video_stream" "default" {
  name = "terraform-kinesis-video-stream-test-%d"

  tags = {
    name = "yak-stream"
  }
}
```

### Debug Output

```
TestAccAWSKinesisVideoStream_basic: resource_aws_kinesis_video_stream_test.go:23: Step 1/3 error: Error running apply: 2020/10/27 13:18:25 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: Error creating Kinesis Video Stream: RequestError: send request failed
        caused by: Post "https://kinesisvideo.us-gov-west-1.amazonaws.com/createStream": dial tcp: lookup kinesisvideo.us-gov-west-1.amazonaws.com on 192.168.22.2:53: no such host

--- FAIL: TestAccAWSKinesisVideoStream_basic (25.23s)
```

### References

* #15869

--- a/aws/resource_aws_kinesis_video_stream_test.go
+++ b/aws/resource_aws_kinesis_video_stream_test.go
@@ -21,7 +21,7 @@ func TestAccAWSKinesisVideoStream_basic(t *testing.T) {
 	rInt2 := acctest.RandInt()
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(kinesisvideo.EndpointsID, t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckKinesisVideoStreamDestroy,
 		Steps: []resource.TestStep{
@@ -62,7 +62,7 @@ func TestAccAWSKinesisVideoStream_options(t *testing.T) {
 	rName2 := acctest.RandString(8)
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(kinesisvideo.EndpointsID, t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckKinesisVideoStreamDestroy,
 		Steps: []resource.TestStep{
@@ -103,7 +103,7 @@ func TestAccAWSKinesisVideoStream_Tags(t *testing.T) {
 	rInt := acctest.RandInt()
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(kinesisvideo.EndpointsID, t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckKinesisVideoStreamDestroy,
 		Steps: []resource.TestStep{
@@ -148,7 +148,7 @@ func TestAccAWSKinesisVideoStream_disappears(t *testing.T) {
 	rInt := acctest.RandInt()
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
+		PreCheck:     func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(kinesisvideo.EndpointsID, t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckKinesisVideoStreamDestroy,
 		Steps: []resource.TestStep{

